pkg/interactor/cloud.go

```diff
@@ -0,0 +1,41 @@
+package interactor
+
+import (
+	"errors"
+
+	"github.com/CESARBR/knot-cloud-storage/pkg/network/cloud"
+)
+
+var ErrTokenMissing = errors.New("no access token provided")
+
+// CloudInteractor ...
+type CloudInteractor struct {
+	cloud cloud.Cloud
+}
+
+// NewCloudInteractor ...
+func NewCloudInteractor(cloud cloud.Cloud) CloudInteractor {
+	return CloudInteractor{cloud}
+}
+
+// Authenticate ...
+func (ci *CloudInteractor) Authenticate(token string) error {
+	token, err := validateToken(token)
+	if err != nil {
+		return err
+	}
+
+	err = ci.cloud.Authenticate(token)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func validateToken(token string) (string, error) {
```

Comment:
Rename this as the name is redundant. :curly_loop: 
```suggestion
func validate(token string) (string, error) {
```

Reply:
Done!